controllers/suite_test.go

```diff
@@ -53,15 +63,59 @@ var (
 	ctx                  context.Context
 	cancel               context.CancelFunc
 	reconciler           *BtpOperatorReconciler
+	updateCh             chan resourceUpdate = make(chan resourceUpdate, 1000)
 )
 
+type resourceUpdate struct {
+	Cr     *v1alpha1.BtpOperator
+	Action string
+}
+
+func resourceUpdateHandler(obj any, t string) {
+	if cr, ok := obj.(*v1alpha1.BtpOperator); ok {
+		updateCh <- resourceUpdate{Cr: cr, Action: t}
+	}
+}
+
+func matchState(state types.State) gomegatypes.GomegaMatcher {
+	return MatchFields(IgnoreExtras, Fields{
+		"Action": Equal(resourceUpdated),
+		"Cr": PointTo(MatchFields(IgnoreExtras, Fields{
+			"Status": MatchFields(IgnoreExtras, Fields{
+				"State": Equal(state),
+			}),
+		})),
+	})
+}
+
+func matchReadyCondition(state types.State, status metav1.ConditionStatus, reason Reason) gomegatypes.GomegaMatcher {
+	return MatchFields(IgnoreExtras, Fields{
+		"Action": Equal(resourceUpdated),
+		"Cr": PointTo(MatchFields(IgnoreExtras, Fields{
+			"Status": MatchFields(IgnoreExtras, Fields{
+				"State": Equal(state),
+				"Conditions": ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":   Equal(ReadyType),
+					"Reason": Equal(string(reason)),
+					"Status": Equal(status),
+				}))),
+			}),
+		})),
+	})
+}
+
+func matchDeleted() gomegatypes.GomegaMatcher {
+	return MatchFields(IgnoreExtras, Fields{"Action": Equal(resourceDeleted)})
+}
+
 func TestAPIs(t *testing.T) {
 
 	RegisterFailHandler(Fail)
 
 	suiteCfg, reporterCfg := GinkgoConfiguration()
 	//Example of usage label:
 	//suiteCfg.LabelFilter = "(test-update)"
+	SetDefaultEventuallyTimeout(time.Second * 5)
```

Comment:
previously, tests configured the same 5 second timeout for each `Eventually()` call, I think it makes sense to set this globally then